custom_components/tahoma/lock.py

```diff
@@ -46,49 +43,44 @@ def update(self):
             return
 
         self.controller.get_states([self.tahoma_device])
-        self._battery_level = self.tahoma_device.active_states["core:BatteryState"]
-        self._name = self.tahoma_device.active_states["core:NameState"]
-        if (
-            self.tahoma_device.active_states.get("core:LockedUnlockedState")
-            == STATE_LOCKED
-        ):
-            self._lock_status = STATE_LOCKED
-        else:
-            self._lock_status = STATE_UNLOCKED
-        self._available = (
-            self.tahoma_device.active_states.get("core:AvailabilityState")
-            == "available"
-        )
 
     def unlock(self, **kwargs):
         """Unlock method."""
-        _LOGGER.debug("Unlocking %s", self._name)
+        _LOGGER.debug("Unlocking %s", self.name)
         self.apply_action("unlock")
 
     def lock(self, **kwargs):
         """Lock method."""
-        _LOGGER.debug("Locking %s", self._name)
+        _LOGGER.debug("Locking %s", self.name)
         self.apply_action("lock")
 
     @property
     def name(self):
         """Return the name of the lock."""
-        return self._name
+        return self.tahoma_device.active_states["core:NameState"]
```

Comment:
def name() could be inherited from tahoma_device. NameState is also not the correct one to use, since this could be truncated for longer labels. Correct one is to use `label`.

Reply:
Done.